src/popup/utils/constants.ts

```diff
@@ -68,6 +72,69 @@ export const TX_FUNCTIONS = {
   claim: 'claim',
 } as const;
 
+export const TX_FUNCTION_TYPE_MULTISIG = {
+  propose: 'propose',
+  confirm: 'confirm',
+  refuse: 'refuse',
+  revoke: 'revoke',
+} as const;
+
+export const TX_FUNCTION_TYPE_DEX: Record<DexFunctionType, TxFunctionRaw[]> = {
```

Comment:
```suggestion
export const DEX_TX_FUNCTIONS: Record<DexFunctionType, TxFunctionRaw[]> = {
```

Reply:
I don't understand this suggestions. In my opinion it is nice to name things starting from more general part and finish with specific. This way when using constants it looks more grouped:
```
TX_FUNCTION_TYPE_DEX
TX_FUNCTION_TYPE_FOO
TX_FUNCTION_TYPE_BAR
```

Reply:
It's look good this way where you defined them but not so good at where you are using it
For eg:
```js
TX_FUNCTION_TYPE_DEX.contains('tip');

vs 

DEX_TX_FUNCTIONS.contains('tip') 

```

later one is more readable.

Reply:
Let me know what you think about this. The existing one is fine too.

Reply:
I was asked for my point of view on that matter so here are my two cents: 
1. If there is more than one variable holding alternative content of the same type, I usually call it [SOME_NAME]_[MODIFIER] so when I put the first part of the const, the IDE suggests all the alternatives I can use without me memorizing what the right [modifier] is called. e.g. 
```
MY_TOYS_CARS = []
MY_TOYS_BLOCKS = []
```
2.  if there is only one variable without any variants then I just follow the English grammar which would be for example `MY_TOY_CARS = []`

---
I think here the first case applies so I would probably go with `TX_FUNCTION_TYPE_DEX` since there are multiple TX_FUNCTION_TYPE variants

Reply:
I created also a poll on our chat and 4 out of 4 available people voted for the new naming convention. I'm closing this.